nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/OsCalls.java

```diff
@@ -0,0 +1,230 @@
+   }
+
+   public Port getPort(String portId) {
+       return this.osClient.networking().port().get(portId);
+   }
+
+   public FlowClassifier updateFlowClassifier(String flowClassifierId, FlowClassifier flowClassifier) {
+       checkArgument(flowClassifierId != null, "null passed for %s !", "Flow Classifier Id");
+       checkArgument(flowClassifier != null, "null passed for %s !", "Flow Classifier");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       flowClassifier = flowClassifier.toBuilder().id(null).projectId(null).build();
+
+       return this.osClient.sfc().flowclassifiers().update(flowClassifierId, flowClassifier);
+   }
+
+   public PortChain updatePortChain(String portChainId, PortChain portChain) {
+       checkArgument(portChainId != null, "null passed for %s !", "Port Chain Id");
+       checkArgument(portChain != null, "null passed for %s !", "Port Chain");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portChain = portChain.toBuilder().id(null).projectId(null).chainParameters(null).chainId(null).build();
+
+       portChain = this.osClient.sfc().portchains().update(portChainId, portChain);
+       return initializePortChainCollections(portChain);
+   }
+
+   public PortPairGroup updatePortPairGroup(String portPairGroupId, PortPairGroup portPairGroup) {
+       checkArgument(portPairGroupId != null, "null passed for %s !", "Port Pair Group Id");
+       checkArgument(portPairGroup != null, "null passed for %s !", "Port Pair Group");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portPairGroup  = portPairGroup.toBuilder().id(null).projectId(null).portPairGroupParameters(null).build();
+
+       portPairGroup = this.osClient.sfc().portpairgroups().update(portPairGroupId, portPairGroup);
+       return portPairGroup;
+   }
+
+   public PortPair updatePortPair(String portPairId, PortPair portPair) {
+       checkArgument(portPairId != null, "null passed for %s !", "Port Pair Id");
+       checkArgument(portPair != null, "null passed for %s !", "Port Pair");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portPair = portPair.toBuilder().id(null).projectId(null).build();
+       portPair = this.osClient.sfc().portpairs().update(portPairId, portPair);
+       return portPair;
+   }
+
+   public Port updatePort(Port port) {
+       return this.osClient.networking().port().update(port);
+   }
+
+   public void deleteFlowClassifier(String flowClassifierId) {
+       ActionResponse response = this.osClient.sfc().flowclassifiers().delete(flowClassifierId);
+       if (!response.isSuccess()) {
+           if (response.getCode() == 404) {
+               return;
+           }
+           String msg = String.format("Opentack returns exception deleting port pair group %s: %s", flowClassifierId, response.toString());
```

Comment:
we should not use toString on response.

extract faultcode and getFault() to construct a clear message.

Would we want to create a custom exception to indicate these errors are coming from external systems?